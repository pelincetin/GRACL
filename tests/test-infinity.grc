void printf(double d){
    print(doubleToString(d));
}
void printb(bool b){
    if (b) printi(1);
    else printi(0);
}
double inf = infinity;
int main() {   
    double inity = infinity;
    printf(inf);
    printf(inity);
    printf(inf + 4.0);
    printf(inf - 2.5);
    printf(inf * 3.9);
    printf(inf / 1.8);
    printf(inf + inity);
    printf(inf - inity);
    printf(inf * inity);
    printf(inf / inity);
    printf(1.0 / inf);
    printf(1.0 / 0.0);
    printf(1.0 - inf);
    printf(-inf);
    printb(inf == 4.0);
    printb(inf == inity);
    printb(4.0 == inf);
    printb(inf != 42.0);
    printb(inf != inity);
    printb(42.0 != inf);
    printb(inf > 10.0);
    printb(inf > inity);
    printb(10.0 > inf);
    printb(inf < 10.0);
    printb(inf < inity);
    printb(10.0 < inf);
    printb(inf >= 10.0);
    printb(inf >= inity);
    printb(10.0 >= inf);
    printb(inf <= 10.0);
    printb(inf <= inity);
    printb(10.0 <= inf);
    return 0;
}