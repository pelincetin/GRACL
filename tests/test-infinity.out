inf
inf
inf
inf
inf
inf
inf
-nan
inf
-nan
0.000000
inf
-inf
-inf
0
1
0
1
0
1
1
0
0
0
0
1
1
1
0
0
1
1